src/index.js

```diff
@@ -6,27 +6,37 @@ export default Selector(complexSelector => {
             throw new Error(`If the selector parameter is passed it should be a string, but it was ${typeof selector}`);
     }
 
-    function validateVueVersion () {
-        const SUPPORTED_VUE_VERSION = 2;
-        const vueVersion            = parseInt(window.Vue.version.split('.')[0], 10);
+    function validateVueVersion (rootInstance) {
+        const MAJOR_SUPPORTED_VUE_VERSION = 2;
+        const vueVersion                  = parseInt(findVueConstructor(rootInstance).version.split('.')[0], 10);
 
-        if (vueVersion < SUPPORTED_VUE_VERSION)
+        if (vueVersion < MAJOR_SUPPORTED_VUE_VERSION)
             throw new Error('testcafe-vue-selectors supports Vue version 2.x and newer');
     }
 
-    function findFirstRootInstance () {
-        let instance     = null;
-        const treeWalker = document.createTreeWalker(document, NodeFilter.SHOW_ELEMENT, () => NodeFilter.FILTER_ACCEPT, false);
-        let currentNode  = treeWalker.nextNode();
+    /*eslint-disable no-unused-vars, no-eval*/
+    function findVueConstructor (rootInstance) {
+        // NOTE: babel converts 'Object.getPrototypeOf' to '(0, _getPrototypeOf2.default)'
```

Comment:
Change comment to
`Testcafe does not support a ClientFunction containing polyfilled functions. See list in https://github.com/babel/babel/blob/master/packages/babel-plugin-transform-runtime/src/definitions.js.
This is why, we use this hack.`